### ISSUE TITLE
DomiqBase (LCN) Binding

### ISSUE THREAD

**Issue by etobi** [openhab/openhab2-addons#661]
I'm running an LCN based HomeAutomation. But as I'm using a Domiq Base (https://domiq.de/products/d_bl_1b) instead of the LCN-PCHK I can't use the existing LCN Binding (https://github.com/openhab/openhab/wiki/LCN-Binding).

The Domiq Base offers a TCP port (4224) and simple protocol to control the LCN. Someone already implemented something using the TCP binding and special rules configuration. I dont feel like this is very sophisticated.

I'll try to implement a native, basic openhab2 binding using the Domiq Base.
##

**Comment by teichsta** [comment]
Thanks for getting in touch @etobi. As this doesn't seem to an issue but more a question i'd like to forward you to the openHAB forum at https://community.openhab.org. You'll most probably find others to discuss your question there.

**Comment by etobi** [comment]
This issue was not intended as a question (no "?" in here ;-) ), but more like an announcement I'm going to work on this. From the contribution guide I thought this is the way to go.

Thanks @kaikreuzer for reopening.

**Comment by teichsta** [comment]
i discussed with Kai already … sorry for getting you wrong here!

**Comment by etobi** [comment]
<3

**Comment by etobi** [comment]
just want to let you know: still a thing and i'm still working on it. however, lack of time..

**Comment by kobelka** [comment]
Hi etobi,
are you still working on it ?
Regards 
Jörg

**Comment by etobi** [comment]
Hej Jörg,

funny you're asking as I was about to write a short update here like very soon.

TL;DR
I'm currently *not* working on this task anymore. I found MQTT which looks like it could solve all of my current home automation issues.


I realised it does not make sense for me to work on this for a couple of reasons.
First of all: It been a lot of years I did java the last time - so I'm very slowly and (this is reason two) didn't find the time to get on track. I did spend a lot of time for setup, understanding the basics of java and the openHAB framework. Whenever I found some hours to work on it, it was very frustrating as the results keep coming veeery solwly - even though the task itself is not very hard and should be easy to implement - for an experienced java developer. I didnt managed to get it solved for the last like 9 month - so i think I wont in the very near future.
But most of all: I found another way to solve my problem (using openHAB or any other home automation) with LCN/Domiq.
Last week I stumbled over home-assistant.io (which BTW also *do not* support LCN/Domiq). A friend of me played around with HA together with a ESP8266 and used MQTT (http://mqtt.org/) for communication. MQTT is a very lightweight protocol for e.g. sensor data and many others (including HA and openHAB *do* support MQTT). As the idea and infrastructure of MQTT might solve a couple of other issues I have (e.g. collecting several sensor data from a handful of raspberry and arduinos at a central hub) I started to write a very simple Domiq/MQTT bridge.

Its talking to Domiq's port 4224 on one side and using a MQTT client on the other end talking to a mosquitto MQTT broker. HA, and even openHAB connect to the mosquitto, too and thus can talk to and read from LCN (over domiq 4224 over mqtt).
Using this setup I can set/trigger anything I can do with the port 4224 on the Domiq base AND I get mqtt messages/events back for every change on the Domiq/LCN.

The brigde is based on node.js (which I'm much more comfortable to write in) and available here:
https://github.com/etobi/node-domiqMqtt

Currently its a very early (although promising) proof-of-concept and very subject to change. I have a couple of additional features in mind - however basically it working.
Also there is no documentation yet.

Feel free to test it and don't hesitate to contact me, if you need help.

kind regards
   tobias

P.S.: If someone would like to pick up this task: I'm sorry I can't even provide any code to start with. All I accomplish was more like getting into the framework and understanding where to hook into.

**Comment by kaikreuzer** [comment]
Thanks for the update @etobi. I close this issue then.
